hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java

```diff
@@ -2485,7 +2485,7 @@ private CredentialEntry getCredentialEntry(CredentialProvider provider,
    * @param name
    * @return clear text password or null
    */
-  protected char[] getPasswordFromConfig(String name) {
+  public char[] getPasswordFromConfig(String name) {
```

Comment:
is there any risk of recursive store init going on?

Reply:
when I used conf.getPassword(), it went to an infinite recursion (init keystore -> go to credential provider -> init keystore ...), that's how I ended up with getPasswordFromConfig

Reply:
aah. filesytems are meant to undeclare themselves as a source of credentials to avoid this...was this secrets for an fs/object store you were trying to retrieve?